aiida_lammps/workflows/base.py

```diff
@@ -0,0 +1,274 @@
+"""Workchain to run a LAMMPS calculation with automated error handling and restarts."""
+import re
+
+from aiida import orm
+from aiida.common import AttributeDict, InputValidationError
+from aiida.engine import (
+    BaseRestartWorkChain,
+    ProcessHandlerReport,
+    process_handler,
+    while_,
+)
+import numpy as np
+
+from aiida_lammps.calculations.base import LammpsBaseCalculation
+from aiida_lammps.utils import RestartTypes
+
+
+class LammpsBaseWorkChain(BaseRestartWorkChain):
+    """Base workchain for calculations using LAMMPS"""
+
+    _process_class = LammpsBaseCalculation
+
+    @classmethod
+    def define(cls, spec):
+        """Define the process specification."""
+        # yapf: disable
+        super().define(spec)
+        spec.expose_inputs(LammpsBaseCalculation)
```

Comment:
Should definitely use a namespace here, because now you are replacing the `metadata` inputs of the workchain with those of the `LammpsBaseCalculation`.
```suggestion
        spec.expose_inputs(LammpsBaseCalculation, namespace='lammps')
```